src/map.cpp

```diff
@@ -6506,6 +6583,19 @@ void map::loadn( const int gridx, const int gridy, const int gridz, const bool u
             return;
         }
     }
+    // Adding snow to new submaps
+    for( int j = 0; j < SEEY; j++ ) {
+        for( int i = 0; i < SEEX; i++ ) {
+            const tripoint pnt( gridx * SEEX + i, gridy * SEEY + j, gridz );
+            const point p( i, j );
+            if( ( !tmpsub->get_ter( p ).obj().has_flag( TFLAG_INDOORS ) ) &&
+                ( !tmpsub->get_ter( p ).obj().has_flag( "TREE" ) )  &&
+                ( tmpsub->get_furn( p ) == f_null ) && ( is_outside( pnt ) ) &&
+                x_in_y( percent_settled, 100 ) ) {
+                tmpsub->frn[i][j] = furn_id( "f_snow" );
```

Comment:
Please use `set_furn()` instead. It has an important side-effect (`is_uniform = false;`). Low-level code is best avoided here.